workspace/__ardulib__/Graphics/XGraphics.h

```diff
@@ -0,0 +1,40 @@
+
+#ifndef X_GRAPHICS_H
+#define X_GRAPHICS_H
+
+#include <Arduino.h>
+
+#include "XRender.h"
+#include "XVector2.h"
+
+using xod::XColor;
+
+struct BBox {
+    XVector2<int16_t> pivot;
+    int16_t width;
+    int16_t height;
+};
+
+class XGraphics {
+protected:
+    XGraphics* _parent;
+
+public:
+    XGraphics();
+    XGraphics(XGraphics* parent);
+
+    virtual XColor getFGColor() const;
+    virtual BBox getUpstreamCanvasBBox() const;
+
+    virtual void renderScanline(XRenderer* renderer, int16_t scanline, uint16_t* buffer, size_t bufferSize) {}
+    void renderScanlineRecursively(XRenderer* renderer, int16_t scanline, uint16_t* buffer, size_t bufferSize);
+    void render(XRenderer* renderer);
+};
+
+inline uint16_t xcolorTo565(XColor color) {
+    return ((color.r & 0xF8) * 256) | ((color.g & 0xFC) * 8) | (color.b / 8);
```

Comment:
In this case `<<`s make much  more sense than multiplications because you are tweaking bits'n'bytes

```suggestion
    return ((color.r & 0xF8) << 8) | ((color.g & 0xFC) << 3) | (color.b >> 1);
```